db/migrate/20200607110209_fix_latest_flag_for_individual_target_submissions.rb

```diff
@@ -0,0 +1,47 @@
+class FixLatestFlagForIndividualTargetSubmissions < ActiveRecord::Migration[6.0]
+  class TimelineEvent < ApplicationRecord
+    has_many :timeline_event_owners
+    belongs_to :target
+    has_many :target_evaluation_criteria, through: :target
+  end
+
+  class TimelineEventOwner < ApplicationRecord
+    belongs_to :timeline_event
+    belongs_to :founder
+  end
+
+  class Founder < ApplicationRecord
+    has_many :timeline_event_owners
+    has_many :timeline_events, through: :timeline_event_owners
+  end
+
+  class Target < ApplicationRecord
+    ROLE_TEAM = 'team'
+
+    has_many :target_evaluation_criteria
+
+    def team_target?
+      role == ROLE_TEAM
+    end
+  end
+
+  class TargetEvaluationCriterion < ApplicationRecord
+  end
+
+  def up
+    Founder.joins(timeline_events: { target: :target_evaluation_criteria }).distinct.find_each do |student|
+      student.timeline_events.joins(target: :target_evaluation_criteria).distinct.group_by(&:target_id).each do |_target_id, submissions|
+        latest_submission = submissions.sort_by { |s| s.created_at }.last
+
+        next if latest_submission.target.team_target?
```

Comment:
Given that all submissions are grouped by target ID, it should be safe to check the target's role _first_ using any of the submissions, before sorting the submissions to get the latest one.

```suggestion
        next if submissions.first.target.role == 'team'

        latest_submission = submissions.sort_by { |s| s.created_at }.last        
```